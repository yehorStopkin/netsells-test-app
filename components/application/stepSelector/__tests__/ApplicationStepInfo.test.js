const { shallowMount } = require('@vue/test-utils');
const ApplicationStepInfo = require('@/components/application/stepSelector/ApplicationStepInfo').default;

describe('ApplicationStepInfo', () => {
    getData = (stepNumber = 0, filingPercentage = 0, isCompleted = false) => {
        return {
            stepNumber,
            filingPercentage,
            isCompleted,
            title: 'Test title',
            subtitle: 'Test subtitle',
            details: 'Test details',
        };
    };

    test('component default', () => {
        const wrapper = shallowMount(ApplicationStepInfo, { propsData: { data: getData() } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();

        const rootElement = wrapper.findAll('.application-step-info');
        expect(rootElement.length).toBe(1);

        const progressBar = wrapper.findAll('.application-step-info__progress-bar');
        expect(progressBar.length).toBe(1);

        const titleElement = wrapper.findAll('.application-step-info__title');
        expect(titleElement.length).toBe(1);

        const subtitleElement = wrapper.findAll('.application-step-info__subtitle');
        expect(subtitleElement.length).toBe(1);

        const detailsElement = wrapper.findAll('.application-step-info__details');
        expect(detailsElement.length).toBe(1);

        const actionElement = wrapper.findAll('.application-step-info__action');
        expect(actionElement.length).toBe(2);
    });

    test('check actionText of completed step', () => {
        const wrapper = shallowMount(ApplicationStepInfo, { propsData: { data: getData(0, 0, true) } });

        const actionTextElement = wrapper.findAll('.application-step-info__action-text');
        expect(actionTextElement.length).toBe(1);
        expect(actionTextElement.at(0).text()).toBe(wrapper.vm.doneText);
    });

    test('check actionText of uncompleted step', () => {
        const wrapper = shallowMount(ApplicationStepInfo, { propsData: { data: getData() } });

        const actionTextElement = wrapper.findAll('.application-step-info__action-text');
        expect(actionTextElement.length).toBe(1);
        expect(actionTextElement.at(0).text()).toBe(wrapper.vm.completeText);
    });

    test('check select step', () => {
        const wrapper = shallowMount(ApplicationStepInfo, { propsData: { data: getData(2) } });

        const actionTextElement = wrapper.findAll('.application-step-info__action-text');
        actionTextElement.at(0).trigger('click');

        expect(wrapper.emitted().selected[0][0]).toEqual(wrapper.vm.data.stepNumber);
    });

    test('check progress getter', () => {
        const wrapper = shallowMount(ApplicationStepInfo, { propsData: { data: getData() } });
        expect(wrapper.vm.progress).toBe(wrapper.vm.data.filingPercentage);
    });
});
