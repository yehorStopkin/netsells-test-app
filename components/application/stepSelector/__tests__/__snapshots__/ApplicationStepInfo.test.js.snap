// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationStepInfo component default 1`] = `
<div
  class="application-step-info"
>
  <div
    class="application-step-info__wrapper"
  >
    <progress-bar-stub
      class="application-step-info__progress-bar"
      progress="0"
    />
     
    <div
      class="application-step-info__title"
    >
      Test title
    </div>
     
    <div
      class="application-step-info__subtitle"
    >
      
            Test subtitle
        
    </div>
     
    <div
      class="application-step-info__details"
    >
      Test details
    </div>
  </div>
   
  <div
    class="application-step-info__action"
  >
    <div
      class="application-step-info__action"
    >
      <div
        class="application-step-info__action-text"
      >
        
                COMPLETE
            
      </div>
       
      <div
        class="application-step-info__action_arrow-icon"
      />
    </div>
  </div>
</div>
`;
